core-bundle/src/Cron/MessengerCron.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCronJob;
+use Contao\CoreBundle\Util\ProcessUtil;
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Promise\Utils;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+
+#[AsCronJob('minutely')]
+class MessengerCron
+{
+    /**
+     * @param array<array{'options': array<string>, 'transports': array<string>, 'autoscale': array{'enabled': bool, 'desired_size': int, 'max': int}}> $workers
+     */
+    public function __construct(private ContainerInterface $messengerTransportLocator, private ProcessUtil $processUtil, private string $consolePath, private array $workers)
+    {
+    }
+
+    public function __invoke(string $scope): PromiseInterface|null
+    {
+        if (Cron::SCOPE_CLI !== $scope) {
+            return null;
+        }
+
+        $workerPromises = [];
+
+        foreach ($this->workers as $worker) {
+            $this->addWorkerPromises($worker, $workerPromises);
+        }
+
+        return Utils::all($workerPromises);
+    }
+
+    /**
+     * @param array{'options': array<string>, 'transports': array<string>, 'autoscale': array{'enabled': bool, 'desired_size': int, 'max': int}} $worker
+     */
+    private function addWorkerPromises(array $worker, array &$workerPromises): void
+    {
+        // Always add one worker
+        $workerPromises[] = $this->createProcessPromiseForWorker($worker);
+
+        if ($worker['autoscale']['enabled']) {
+            $totalMessages = $this->collectTotalMessages($worker['transports']);
+            $desiredWorkers = ceil($totalMessages / $worker['autoscale']['desired_size']);
+
+            // Never more than the max (-1 because we've already started one)
```

Comment:
```suggestion
            // Never more than the max
```
this is done two lines below with its own comment.